### PR TITLE
chore: topics for all FlutterFire plugins in `pubspec.yaml` files

### PR DESCRIPTION
**Pull request by russellwheatley**
## Description

Topics for all plugins.

## Related Issues

closes https://github.com/firebase/flutterfire/issues/12777

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process. Updating the `pubspec.yaml` and changelogs is not required.

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [ ] All existing and new tests are passing.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] The analyzer (`melos run analyze`) does not report any problems on my PR.
- [ ] I read and followed the [Flutter Style Guide].
- [ ] I signed the [CLA].
- [ ] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change.
- [ ] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/firebase/flutterfire/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

**Review by Lyokone — APPROVED** [review]

File: packages/firebase_vertexai/firebase_vertexai/pubspec.yaml [hunk 1]
```diff
@@ -2,6 +2,11 @@ name: firebase_vertexai
 description: "Firebase Vertex AI SDK."
 version: 0.1.1
 homepage: https://firebase.google.com/docs/vertex-ai/get-started?platform=flutter
+topics:
+  - firebase
+  - vertex
```
Comment (cynthiajoan) [inline packages/firebase_vertexai/firebase_vertexai/pubspec.yaml:7]:
```suggestion
  - vertexai
```

File: packages/firebase_vertexai/firebase_vertexai/pubspec.yaml [hunk 2]
```diff
@@ -2,6 +2,11 @@ name: firebase_vertexai
...
+  - ai
```
Comment (cynthiajoan) [inline packages/firebase_vertexai/firebase_vertexai/pubspec.yaml:8]:
```suggestion
  - gemini
```

File: packages/firebase_vertexai/firebase_vertexai/pubspec.yaml [hunk 3]
```diff
@@ -2,6 +2,11 @@ name: firebase_vertexai
...
+  - generative
```
Comment (cynthiajoan) [inline packages/firebase_vertexai/firebase_vertexai/pubspec.yaml:9]:
```suggestion
  - generative-ai
```

**Review by cynthiajoan — APPROVED** [review]
With some suggestions

**Pull request merged**